Source/FieldSolver/SpectralSolver/SpectralKSpace.cpp

```diff
@@ -142,9 +142,14 @@ SpectralKSpace::getSpectralShiftFactor( const DistributionMapping& dm,
             case ShiftType::TransformFromCellCentered: sign = -1.; break;
             case ShiftType::TransformToCellCentered: sign = 1.;
         }
-        constexpr Complex I{0,1};
+        const Complex I{0,1};
         for (int i=0; i<k.size(); i++ ){
+#ifdef AMREX_USE_GPU
+            shift[i] = thrust::exp( I*sign*k[i]*0.5*dx[i_dim] );
+#else
             shift[i] = std::exp( I*sign*k[i]*0.5*dx[i_dim] );
+#endif
```

Comment:
Interesting. Why is it necessary to do `thrust::exp` here, but not `thrust::cos` in the `InitializeSpectralCoefficients` function? Is this related to the `Complex` type that we use on GPU?

Reply:
Yes, it is because of the Complex type that we use is thrust::complex<amrex::Real>, and not due to the exp. 

Here is the compilation error:
error: no instance of overloaded function "std::exp" matches the argument list
            argument types are: (thrust::complex<double>)